core/src/main/java/feign/Capability.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright 2012-2020 The Feign Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package feign;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Arrays;
+import java.util.List;
+import feign.Logger.Level;
+import feign.Request.Options;
+import feign.codec.Decoder;
+import feign.codec.Encoder;
+
+public interface Capability {
+
+  static <E> E enrich(E target, List<Capability> capabilities) {
+    return capabilities.stream()
+        .reduce(
+            target,
+            (c, capability) -> invoke(c, capability),
+            (a, b) -> b);
```

Comment:
This reads like very old `C` code.  Can we please update this to be more specific?  I can't understand what's happening here enough to comment.

Reply:
Yeah, this look odd, I will add some docs explaining.